docs/commands/lint.md

```diff
@@ -5,24 +5,55 @@ title: modular lint
 
 # `modular lint [options] [regexes...]`
 
-`modular lint` will check the diff between the current branch and your remote
-origin default branch (i.e. `master` or `main`) and only lint the `.ts`, `.tsx`,
-`.js`, `.jsx` files that have changes.
+`modular lint` without arguments or options will check the diff between the
+current branch and your remote origin default branch (i.e. `master` or `main`)
+and only lint the `.ts`, `.tsx`, `.js`, `.jsx` files that have changes. When in
+CI or when `--all` is provided, it will default to lint the entire codebase.
+Wneh `--staged` is provided, it will only lint the files staged on git.
+
+Alternatively, when it is invoked with one or more selective options
+(`--packages`, `--ancestors`, `--descendants` or `--changed`), it will lint all
+the source code files in the provided `--packages`, optionally agumenting the
+set of packages by adding their ancestors (`--ancestors`), their descendants
+(`--descendants`) or the files that have changed (`--changed`), calculated
+comparing the current state of the git repository with the branch specified by
+`compareBranch` or, if `compareBranch` is not set, with the default branch.
 
 It uses your project eslint config to lint and jest runner to report on the lint
 status of each file (`PASS` or `FAIL`). If a file has a lint warning, the
 command will be marked as failed, with printed eslint errors for each
 warning/error.
 
-When in CI, it will default to lint the entire codebase. When not in CI, it will
-default to only lint the diffed files.
-
 ## Options:
 
-`--all`: Lints the entire codebase
+`--all`: Lints the entire codebase (not compatible with `--staged` and with the
```

Comment:
Related, do you think typecheck should have the `--all` option, too?

Reply:
No. `--all` is here only because removing it would be breaking. Typecheck does the right thing already (typechecks the whole project when no options are specified)